### PR TITLE
Swift 4 Decodable pattern for basic queries

### PR DESCRIPTION
**Pull request by patrick-zippenfenig**
```swift
struct User: Decodable {
  let id: Int
  let user_name: String
  let age: Int?
}
let rows: [User] =  try conn.query("SELECT id, user_name FROM users WHERE id = 123")
```
See https://github.com/novi/mysql-swift/issues/58. QueryRowResultType still supported.

### CODE REVIEW

File: Sources/MySQL/Result.swift [hunk 1]
```diff
@@ -142,3 +142,120 @@ public struct QueryRowResult {
...
+    public func unkeyedContainer() throws -> UnkeyedDecodingContainer {
+        throw QueryError.initializationError
+    }
+    
+    public func singleValueContainer() throws -> SingleValueDecodingContainer {
+        throw QueryError.initializationError
+    }
+}
+
+struct RowKeyedDecodingContainer<K : CodingKey> : KeyedDecodingContainerProtocol {
```
Comment (novi) [inline Sources/MySQL/Result.swift]: @patrick-zippenfenig We could change `struct`  to `fileprivate struct`.

**Comment by novi** [comment]
Thanks for the pull request.
The API design looks good.

Could you write a test for this decodable support?

**Comment by patrick-zippenfenig** [comment]
@novi Basic tests seem to be fine. I implemented a first version of "Encodable" locally. It's a little bit more complicated then Decodable. I will start another PR for encoding

**Comment by novi** [comment]
@patrick-zippenfenig Thanks and the test is fine.
I noticed that `AutoincrementID` is not yet decodable and `Data` (Blob) test is absent.
But I will merge this PR for now. We could open another PR for that.

**Pull request merged**